### PR TITLE
Profiles: Add zoomable image preview to ENS cover photos

### PR DESCRIPTION
**Pull request by jxom**
Fixes RNBW-2877

## What changed (plus any additional context for devs)

This PR adds support for zoomable ENS cover photos in the NFT expanded state.

The `ZoomableWrapper` component is quite specific to the NFT expanded state image, so I've had to tweak it to be more generalised.

As a part of this, I have introduced a new component called `ImagePreviewOverlay` which will allow us to use the zoomable functionality anywhere (it is still needs a bit of tweaking to handle multiple images, but is OK for now for the ENS cover photos case).

Idea is that we wrap a screen with `ImagePreviewOverlay` (this is required to position the image correctly on the screen in the zoom state):

```jsx
function Screen() {
  return (
    <SlackSheet>
      <ImagePreviewOverlay>
        ...
      </ImagePreviewOverlay>
    </SlackSheet>
  )
}
```

and then somewhere in our child components, we can add a `ImagePreviewOverlayTarget` component with an image `uri` prop (which will provide the appropriate bounds/positioning for `ImagePreviewOverlay` to render the zoomable image):

```jsx
function ExampleInfoSection() {
  return (
    ...
      <ImagePreviewOverlayTarget uri={uri} />
    ...
  )
}
```

In the PoW video below, the `ImagePreviewOverlayTarget` is placed where the cover image is rendered (basically, if you want zoomable functionality, use `ImagePreviewOverlayTarget` instead of `ImgixImg`).

## PoW (screenshots / screen recordings)

https://www.loom.com/share/a27cdd0cd4664016b142089044020f3c

### CODE REVIEW

File: src/components/expanded-state/UniqueTokenExpandedState.tsx [hunk 1]
```diff
@@ -378,244 +383,254 @@ const UniqueTokenExpandedState = ({
...
-                    <L2Disclaimer
-                      assetType={AssetTypes.polygon}
-                      colors={colors}
-                      hideDivider
-                      isNft
-                      marginBottom={0}
-                      marginHorizontal={0}
-                      onPress={handleL2DisclaimerPress}
-                      symbol="NFT"
+        <ImagePreviewOverlay
```
Comment (jxom) [inline src/components/expanded-state/UniqueTokenExpandedState.tsx:386]: The only diff here is that I wrapped the children w/ `ImagePreviewOverlay`

**Review by pugson — APPROVED** [review]
sick 🔥

**Review by osdnk — CHANGES_REQUESTED** [review]
Please, be v careful while reviewing this component and always ask for QA

Android:
1. `develop` with fixes from #3114
https://streamable.com/ndvts0 <- observe that this is not possible to interact w/ scrollview below and dismiss the sheet 
2. your branch
https://streamable.com/y1b8ai <- I can dismiss the sheet and scroll the scrollview below the image 


iOS: correct. I even find your PR fixing the issue I just discovered:
1. `develop` https://streamable.com/jlkvyn <- see I can scroll scrollview below (by touching the area below nft)
2. your branch: https://streamable.com/sax8r9

**Comment by osdnk** [comment]
I ask for QA here just in the context of the Expanded asset state. I am sorry for breaking the flow for profiles. I believe some minor issues might be not noticed when just doing the whole app review.

**Comment by estebanmino** [comment]
@jxom @osdnk we'll definitely do QA when merging  `@jxom/ens-phase-three` to `develop`, we're now merging to that branch. Do you think fixing this bug would be enough to do QA later or better safe than sorry?

**Comment by osdnk** [comment]
Sorry, closing unintended!

**Comment by osdnk** [comment]
Yeah, I was discussing this one day and this component is super fragile and needs a bit of special treatment. I don't expect to make a QA from the perspective of profiles, just to see the chart expanded state. 
I insist on review now, because:
1. while making the holistic review it will be easy to skip that scenario while testing a ton of other features. 
2. I see that I press the red button in 70% of PRs related to this component. So you will be probably blocked anyway if we will not test this carefully now. 
3. I don't think this is just fixing a bug, this is adding some features (making the code reusable in other places). Some changes are modifying components that are "invisible" (e.g., Gesture Blockers) 


One protip for developing - you can try to set random colors to all views that do nothing from a UI perspective and observe if their position did not change.

**Comment by jxom** [comment]
Haha yeah that's a good tip. I did do the blue, green, red, yellow background color trick. Worked nice 😝

But yeah, good point. Might be better if we QA this PR now.

**Review by ibrahimtaveras00 — APPROVED** [review]
Issues Michal mentioned above have been fixed, looks good on both OS's

**Pull request merged**